docs/android/testing/screenshot_testing.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Android screenshot testing"
+sidebar_label: "Screenshot testing"
+---
+
+🚧🚧🚧 Under Construction 🚧🚧🚧
+
+## Why do we perform screenshot testing?
+
+Screenshot testing is used to verify that the current UI matches the reference UI stored in the repository. By doing this, we can ensure that any changes impacting the UI are intentional and validated. The scope of these tests is limited to the UI.
+
+We should test on a variety of device shapes and sizes, ranging from small screens (e.g., Wear OS devices) to very large screens (e.g., 55" TVs).
+
+### Benefits of screenshot testing
+
+- **UI Consistency**: Ensures that the UI remains consistent across updates.
+- **Library Updates**: Validates that updates to UI libraries do not introduce unintended changes.
+- **Broad Device Coverage**: Tests across multiple screen sizes and shapes to ensure compatibility.
+
+### Real-world example
+
+Screenshot testing has proven useful when using beta versions of libraries, such as the Wear Compose library, where changes in the library could impact the UI.
+
+## Compose screenshot testing
+
+We use the [Compose Preview Screenshot Testing](https://developer.android.com/studio/preview/compose-screenshot-testing) framework to assert that the UI does not change unexpectedly.
+
+### Advantages of compose screenshot testing
+
+- **No emulator required**: These tests do not require an emulator, making them less resource-intensive and significantly faster than [integration tests](integration_testing).
+- **Fast feedback**: Developers can quickly verify UI changes without waiting for emulator boot times.
+
+### Reference screenshots
+
+The reference screenshots are stored under `src/debug/screenshotTest/reference` in each Gradle module. To update the reference screenshots, run the following command:
+
+```bash
+./gradlew updateDebugScreenshotTest
+```
+
+### CI integration
+
+Our [CI pipeline](../ci) verifies the test reports for any errors. If discrepancies are found, the CI blocks the pull request until the issues are resolved.
+
+## Avoiding duplication in Compose previews
+
+To avoid duplicating Compose previews in your tests, ensure that you reuse existing composables and preview annotations wherever possible. This reduces redundancy and ensures consistency between previews and tests.
```

Comment:
Maybe a quick example would be helpful here? Would the current test be a good example?